doc/whats_new/v1.0.rst

```diff
@@ -637,6 +637,10 @@ Changelog
   unavailable on the basis of state, in a more readable way.
   :pr:`19948` by `Joel Nothman`_.
 
+- |Enhancement| Raise informative error message in :func:`type_of_target` when
```

Comment:
Changelog entry needs to be moved to 1.3.

The text needs to be updated to reflect the warning.